### PR TITLE
[core] Add validation in Schema#Builder

### PR DESCRIPTION
**Pull request by yuzelin**
### Purpose

Currently, Schema#Builder.build hasn't check the arguments. This PR add validation to it.

### Tests

No.

### API and Format 

No.

### Documentation

No.

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/schema/Schema.java [hunk 1]
```diff
@@ -94,7 +95,22 @@ public String comment() {
     private static List<DataField> normalizeFields(
             List<DataField> fields, List<String> primaryKeys, List<String> partitionKeys) {
         List<String> fieldNames = fields.stream().map(DataField::name).collect(Collectors.toList());
+
+        Set<String> duplicateColumns = duplicate(fieldNames);
```
Comment (JingsongLi) [inline paimon-core/src/main/java/org/apache/paimon/schema/Schema.java:99]: Add a test for `Schema.newBuilder`?

File: paimon-core/src/main/java/org/apache/paimon/schema/Schema.java [hunk 2]
```diff
@@ -133,6 +155,12 @@ private static List<DataField> normalizeFields(
         return newFields;
     }
 
+    private static Set<String> duplicate(List<String> names) {
```
Comment (SteNicholas) [inline paimon-core/src/main/java/org/apache/paimon/schema/Schema.java:158]: Could we return the result that uses `anyMatch(name -> Collections.frequency(names, name) > 1)` that represents whether the `names` is duplicate? Meanwhile uses the return value to invoke `Preconditions.checkState`.

Comment (yuzelin) [reply paimon-core/src/main/java/org/apache/paimon/schema/Schema.java:158]: I collect duplicate columns to put them in error message:
`Found: {duplicatedColumns}`. 
Maybe it is not necessary. What's your suggestion?

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**